buildSrc/src/main/kotlin/org/gradle/plugins/buildscan/BuildScanConfigurationPlugin.kt

```diff
@@ -0,0 +1,209 @@
+                    val errors = codenarc.getElementsByTag("Package").map { codenarcPackage ->
+                        codenarcPackage.getElementsByTag("File").map { file ->
+                            file.getElementsByTag("Violation").map { violation ->
+                                val filePath = project.rootProject.relativePath(file.attr("name"))
+                                "$filePath:${violation.attr("lineNumber")} \u2192 ${violation.getElementsByTag("Message").first().text()}"
+                            }
+                        }.flatten()
+                    }.flatten()
+
+                    errors.forEach { project.buildScan.value("CodeNarc Issue", it) }
+                }
+            }
+        }
+    }
+
+    private
+    fun Project.extractCiOrLocalData() {
+        if (isCiServer) {
+            buildScan.tag("CI")
+            buildScan.tag(System.getenv("TEAMCITY_BUILDCONF_NAME"))
+            buildScan.link("TeamCity Build", System.getenv("BUILD_URL"))
+            buildScan.value("Build ID", System.getenv("BUILD_ID"))
+            setCommitId(System.getenv("BUILD_VCS_NUMBER"))
+        } else {
+            buildScan.tag("LOCAL")
+        }
+    }
+
+    private
+    fun Project.extractVcsData() {
+
+        fun Project.run(vararg args: String): String {
+            val process = ProcessBuilder(args.toList())
+                .directory(rootDir)
+                .start()
+            assert(process.waitFor() == 0)
+            return process.inputStream.bufferedReader().readText().trim()
+        }
+
+        fun Project.execAsync(f: Runnable) {
+            val latch = java.util.concurrent.CountDownLatch(1)
+            thread(start = true) {
+                try {
+                    f.run()
+                } catch (e: Exception) {
+                    rootProject.logger.warn("Build scan user data async exec failed", e)
+                } finally {
+                    latch.countDown()
+                }
+            }
+        }
+
+        execAsync(Runnable {
+            val commitId = run("git", "rev-parse", "--verify", "HEAD")
+            setCommitId(commitId)
+            val status = run("git", "status", "--porcelain")
+            if (!status.isEmpty()) {
+                buildScan {
+                    tag("dirty")
+                    value("Git Status", status)
+                }
+            }
+        })
+
+        execAsync(Runnable {
+            val branchName = run("git", "rev-parse", "--abbrev-ref", "HEAD")
+            if (!branchName.isEmpty() && branchName != "HEAD") {
+                buildScan.tag(branchName)
+                buildScan.value("Git Branch Name", branchName)
+            }
+        })
+    }
+
+    private fun Project.extractBuildCacheData() {
+        if (gradle.startParameter.isBuildCacheEnabled) {
+            buildScan.tag("CACHED")
+
+            val tasksToInvestigateForCacheWithPaths = if (project.hasProperty("cache.investigate.tasks.paths"))
```

Comment:
I guess we can remove this field and not look into the corresponding property any more.